### DISCUSSION TITLE
Is there a way to list all panes?

### DISCUSSION THREAD

**Discussion by tucker-m** [zellij-org/zellij#3330] [Q&A]
Does zellij have a way of listing all panes? I'm thinking of something like `zellij action query-tab-names`, except for panes instead of tabs.

Also, can you tell it to focus on a pane by name? Like, `zellij action go-to-pane <name>`? I don't see that either, but maybe I'm just not searching for the right thing.

**Comment by giyany** [comment]
Maybe these plugin can give you what you're looking for:

https://github.com/Nacho114/harpoon

https://github.com/FuriouZz/zjpane

**Comment by tucker-m** [comment]
Thank you! Those look helpful. Looks like zjpane does part of what I'm looking for. Maybe I can contribute to zjpane and make it print out a list of panes.

IMO, it would be great if these things were core features of zellij. I can think of many use cases where someone would want to get a list of panes, say, to pipe into `fzf`.

**Reply by giyany** [reply]
I hear you. The philosophy of zellij is to invest in a plugin system, encouraging users to develop and share according to their own needs instead of leaving them to relay on core features. This will probably be the answer of the main developers, though personally I'm not familiar enough with the plugin system to give practical advice.
If you end up contributing, feel free to drop into Discord/Matrix for help in development and to tell us about what you've done. That way we can refer others.
